modules/users/client/controllers/settings/manage-social-accounts.client.controller.js

```diff
@@ -7,7 +7,9 @@ angular.module('users').controller('SocialAccountsController', ['$scope', '$http
     // Check if there are additional accounts
     $scope.hasConnectedAdditionalSocialAccounts = function (provider) {
       for (var i in $scope.user.additionalProvidersData) {
-        return true;
```

Comment:
Can you try to fix this to be more logical instead of us just band-aiding it if possible?